opengrok-indexer/src/main/java/org/opengrok/indexer/framework/PluginFramework.java

```diff
@@ -382,7 +380,7 @@ public final void reload() {
 
         // trashing out the old instance of the loaded enables us
```

Comment:
This caught my eye when going through the warnings in #4312. I was wondering why the `doPrivileged()` method was actually needed. Based on the comment I'd assume that it somehow invoked GC to get rid of the old object, based on the *trashing out* sentence to perform some atomic swap, however I think this is not the case.

Rather, it seems to be related to the plugin model as sort of explained on https://stackoverflow.com/a/2243262/11582827 however it is still not clear to me why the constructor would need to be invoked like this. I can see the parent class (`java.lang.ClassLoader`) consulting the `SecurityManager`. Possibly it needs to check the `getClassLoader` permission which might require running the code with higher privileges.

My worry was that the plugin architecture might be dependent on it, to isolate the plugins from the plugin framework.

Reply:
I agree with you regarding the impact on plugin framework, for consumers using java's security manager this change may cause a regression.  
We can undo this modification for the time being, but when we upgrade the minimum required Java version to 17, we must address it and appropriately communicate in release notes.

Reply:
I reckon the removal of the `doPrivileged()` call here is fine. I think the PluginFramework tests have sufficient coverage to catch significant problem. It is just not clear to me why it was needed. Building with JDK 11 without the call in place does not cause any warnings as far as I can tell and the `AuthorizationFrameworkReloadTest` tests pass.

Reply:
We are not seeing any warnings in JDK 11 because the doPrivileged() api is only deprecated in JDK 17 and later.

DoPrivileged will ensure that the permission check is limited to the present code and not performed on the entire stack trace. What the doPrivileged method's effects are explained in detail on page 53 of https://apps.dtic.mil/sti/pdfs/AD1146140.pdf . 
Having said that, I agree with you on removing doProvileged because java security manager is rarely used these days, and I don't see a usecase for having callers of the PluginFramework class from a different protection domain.